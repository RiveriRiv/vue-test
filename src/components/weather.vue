<template>
	<table id="table"><tr><th class="th">Town</th><th class="th">Temperature</th></tr><tr><td class="td">{{town}}</td><td class="td">{{temperature}}</td></tr></table>
</template>
<script>
export default {
	computed: {
        town() {
            return this.$store.state.town
        },
        temperature() {
            return this.$store.state.temperature
        }
    }
	}
</script>