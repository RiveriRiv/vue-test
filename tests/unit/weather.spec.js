import { createLocalVue } from '@vue/test-utils'
import { mount } from "@vue/test-utils";
import weather from "@/components/weather.vue";
import Vuex from 'vuex';

const localVue = createLocalVue()

localVue.use(Vuex)

describe("weather.vue", () => {
  let actions
  let store
  let state

  beforeEach(() => {
	state = {
      town: 'Moscow',
	  temperature: 20,
	  day: 2
    }

    actions = {
      updateTown: jest.fn(),
      updateTemperature: jest.fn()
    }
    store = new Vuex.Store({
	  state,
      actions
    })
  })

  it("is vue instance", () => {
    const wrapper = mount(weather, {store, localVue});
    expect(wrapper.isVueInstance()).toBe(true)
  });

  it("matches snapshot", () => {
    const wrapper = mount(weather, {store, localVue});
     expect(wrapper.html).toMatchSnapshot();
  });

   it("matches data", () => {
    const wrapper = mount(weather, {store, localVue});
     expect(wrapper.vm.town).toBe('Moscow')
	 expect(wrapper.vm.temperature).toBe(20)
  });
});
