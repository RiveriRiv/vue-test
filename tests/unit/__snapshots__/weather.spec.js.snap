// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`weather.vue matches spanpshot 1`] = `[Function]`;
